event_sale_reservation/models/sale_order_line.py

```diff
@@ -28,5 +28,14 @@ class SaleOrderLine(models.Model):
     @api.depends("event_registration_ids")
     def _compute_event_registration_count(self):
         """Get count of related event registrations."""
+        items_data = self.env["event.registration"].read_group(
```

Comment:
There's an ongoing debate about what is more performant lately, as the cache avoids to do constant queries to DB as the `read_group` does, but it also reads more data. If v17 contains the `read_group` option reading from cache, it will be clear. Meanwhile, I think you can leave it as it was for minimizing diff.